File: DSA/Hima-Swarupa-Padala/day15q1.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

def tree_height(root):
    if root is None:
        return -1
    else:
        left_height = tree_height(root.left)
        right_height = tree_height(root.right)
        return max(left_height, right_height) + 1

root = TreeNode(3)
root.left = TreeNode(2)
root.right = TreeNode(5)
root.left.left = TreeNode(1)
root.left.right = TreeNode(4)
root.right.right = TreeNode(6)

print(tree_height(root))
  

File: DSA/Shruti-Pandey/Day_15_q1.java
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    
    TreeNode(int val) {
        this.val = val;
    }
}

public class BinaryTreeHeight {
    public static int treeHeight(TreeNode root) {
        if (root == null) {
            return -1;  
        }
        
        int leftHeight = treeHeight(root.left);
        int rightHeight = treeHeight(root.right);
        
        return Math.max(leftHeight, rightHeight) + 1;
    }
    
    public static void main(String[] args) {
        
        TreeNode root = new TreeNode(3);
        root.left = new TreeNode(1);
        root.right = new TreeNode(4);
        root.right.right = new TreeNode(6);
        
        int height = treeHeight(root);
        System.out.println("The height of the binary tree is: " + height);
    }
}
